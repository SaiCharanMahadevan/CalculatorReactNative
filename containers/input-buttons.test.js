import React from 'react';
import { shallow } from 'enzyme';

import InputButtonsContainer from './input-buttons';

describe('InputButtonsContainer', () => {
  test('renders the section that displays the input operators and operands', () => {
    const wrapper = shallow(
      <InputButtonsContainer onButtonPress={() => {}} onClearLongPressed={() => {}}/>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('calls `onButtonPress` with the relevant value when a button is pressed', () => {
    const wrapper = shallow(
      <InputButtonsContainer onButtonPress={jest.fn((x)=>x)}/>,
    );
    let button = wrapper.find('.Button00').props().onPress('7');
    expect(wrapper.instance().props.onButtonPress).toMatchSnapshot();
  });

  test('calls `onLongPress` with the relevant value when a button is pressed', () => {
    const wrapper = shallow(
      <InputButtonsContainer onLongPress={jest.fn((x)=>x)}/>,
    );
    let button = wrapper.find('.clear-button').props().onLongPress('CLEAR');
    expect(wrapper.instance().props.onButtonPress).toMatchSnapshot();
  });
});
