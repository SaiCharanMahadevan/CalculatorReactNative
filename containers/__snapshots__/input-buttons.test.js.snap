// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputButtonsContainer calls \`onButtonPress\` with the relevant value when a button is pressed 1`] = `[MockFunction]`;

exports[`InputButtonsContainer calls \`onLongPress\` with the relevant value when a button is pressed 1`] = `undefined`;

exports[`InputButtonsContainer renders the section that displays the input operators and operands 1`] = `
<Styled(View)
  alignItems="center"
  className="input-buttons"
  flex={6}
>
  <Styled(View)
    gap={true}
  >
    <Button
      className="clear-button"
      label="CLEAR"
      onLongPress={[Function]}
      onPress={[Function]}
      operator={true}
      width={345}
    />
  </Styled(View)>
  <Styled(Styled(View))>
    <Styled(View)
      gutter={0}
    >
      <Styled(Styled(View))>
        <Styled(View)
          key="0"
        >
          <Button
            className="Button00"
            label="7"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
        <Styled(View)
          key="1"
        >
          <Button
            className="Button01"
            label="8"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
        <Styled(View)
          key="2"
        >
          <Button
            className="Button02"
            label="9"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(View) />
      <Styled(Styled(View))>
        <Styled(View)
          key="0"
        >
          <Button
            className="Button10"
            label="4"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
        <Styled(View)
          key="1"
        >
          <Button
            className="Button11"
            label="5"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
        <Styled(View)
          key="2"
        >
          <Button
            className="Button12"
            label="6"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(View) />
      <Styled(Styled(View))>
        <Styled(View)
          key="0"
        >
          <Button
            className="Button20"
            label="1"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
        <Styled(View)
          key="1"
        >
          <Button
            className="Button21"
            label="2"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
        <Styled(View)
          key="2"
        >
          <Button
            className="Button22"
            label="3"
            onPress={[Function]}
            operand={true}
          />
        </Styled(View)>
      </Styled(Styled(View))>
      <Styled(View) />
      <Styled(Styled(View))>
        <Styled(View)>
          <Button
            label="0"
            onPress={[Function]}
            operand={true}
            width={165}
          />
        </Styled(View)>
        <Styled(View)>
          <Button
            label="="
            onPress={[Function]}
          />
        </Styled(View)>
      </Styled(Styled(View))>
    </Styled(View)>
    <Styled(View)
      gutter={0}
    >
      <Styled(View)>
        <Button
          label="รท"
          onPress={[Function]}
          operator={true}
        />
      </Styled(View)>
      <Styled(View) />
      <Styled(View)>
        <Button
          label="x"
          onPress={[Function]}
          operator={true}
        />
      </Styled(View)>
      <Styled(View) />
      <Styled(View)>
        <Button
          label="+"
          onPress={[Function]}
          operator={true}
        />
      </Styled(View)>
      <Styled(View) />
      <Styled(View)>
        <Button
          label="-"
          onPress={[Function]}
          operator={true}
        />
      </Styled(View)>
      <Styled(View) />
    </Styled(View)>
  </Styled(Styled(View))>
</Styled(View)>
`;
